### PR TITLE
Basset Support

### PR DESCRIPTION
**Pull request by tabacitu**
### BEFORE - What was wrong? What was happening before this PR?

Some files were loaded from CDN, some from local.

### AFTER - What is happening after this PR?

Everything's loaded from local, even though they're pointing to CDNs.

### How did you achieve that, in technical terms?

Use bassets 🐶

### Is it a breaking change or non-breaking change?

Breaking

### How can we test the before & after?

Don't. It's not working fully yet. The remaining problem is inside `common_styles.blade.php`... the last file... the theme... yeah, we have a problem there, because all toolbar icons are actually images. So when bassets tries to load the imagines... it can't, because they're not in the same place 🤦‍♂️

### CODE REVIEW

File: resources/assets/css/elfinder.backpack.theme.css [hunk 1]
```diff
@@ -99,7 +99,7 @@ body #elfinder.ui-widget-content {
 
 /* contextmenu */
 .elfinder-contextmenu .ui-state-hover { background: #3875d7; color:#fff; }
-.elfinder-contextmenu .ui-state-hover .elfinder-contextmenu-arrow { background-image:url('../img/arrows-active.png'); }
+/* .elfinder-contextmenu .ui-state-hover .elfinder-contextmenu-arrow { background-image:url('../img/arrows-active.png'); } */
```
Comment (promatik) [inline resources/assets/css/elfinder.backpack.theme.css:102]: This image was nowhere, and it was causing a warning in console.

File: resources/views/common_scripts.blade.php [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-        <!-- elFinder JS (REQUIRED) -->
-        <script src="{{ asset($dir.'/js/elfinder.min.js') }}"></script>
+        {{-- jQuery UI and Smoothness theme --}}
+        @bassetArchive('https://github.com/jquery/jquery-ui/archive/refs/tags/1.13.2.zip', 'jquery-ui-1.13.2')
+        @basset('jquery-ui-1.13.2/jquery-ui-1.13.2/dist/themes/smoothness/jquery-ui.min.css')
+        @basset('jquery-ui-1.13.2/jquery-ui-1.13.2/dist/jquery-ui.min.js')
 
+        {{-- elFinder JS (REQUIRED) --}}
+        @bassetArchive('https://github.com/Studio-42/elFinder/archive/refs/tags/2.1.61.zip', 'elfinder-2.1.61')
+        @basset('elfinder-2.1.61/elFinder-2.1.61/js/elfinder.min.js')
```
Comment (promatik) [inline resources/views/common_scripts.blade.php:13]: `@bassetArchive()`

File: src/FileManagerServiceProvider.php [hunk 3]
```diff
@@ -39,14 +38,11 @@ protected function bootForConsole()
...
+        // Publishing config file.
         $this->publishes([
-            __DIR__.'/../config/elfinder.php'      => config_path('elfinder.php'),
+            __DIR__.'/../config/elfinder.php' => config_path('elfinder.php'),
         ], 'config');
 
-        $this->publishes([
-            __DIR__.'/../public/packages/backpack/filemanager/themes/Backpack'      => public_path('packages/backpack/filemanager/themes/Backpack'),
-        ], 'public');
-
```
Comment (promatik) [inline src/FileManagerServiceProvider.php:49]: Thanks to basset we no longer need to publish this css 🎉

**Comment by promatik** [comment]
Also, improved the installation console command to be inline with the rest of the ecosystem 👌

image

**Comment by promatik** [comment]
**NOTE**

This dependes on https://github.com/Laravel-Backpack/basset/pull/20

**Comment by tabacitu** [comment]
Tested, works like a charm! 🎉 Thanks Antonio. 
I will merge when v6 gets launched. Until then we can use this branch as the `v6` one.

**Pull request merged**